### DISCUSSION TITLE
Balancing mod score multipliers

### DISCUSSION THREAD

**Discussion by molneya** [ppy/osu#24608] [Ideas]
## The problem
osu!lazer has many new mods and customisation settings compared to the stable client. Many of these additions have default score multipliers (1.0x) despite changing the difficulty of gameplay in some way or highly reduced multipliers to prevent abuse. Therefore changes must to be done to balance these settings so any combination of mod customisation is fair for gameplay/leaderboards. Mod multipliers for customisation settings should also be as easy for the player to understand, and all the complicated intricacies for how it affects gameplay can be left for pp calc.

## Matching stable
As a baseline, the balancing for mods that exist in stable should match. For this I suggest:
**Flashlight:** 1.12x -> 1.10x (matches DoubleTime, for the time being)
**Hidden:** 1.06x -> 1.05x (to balance compared to Flashlight/DoubleTime)
**HardRock:** 1.06x -> 1.05x (match Hidden, for the time being)

There are also mods that provide a similar level of obfuscation to the gameplay as an existing stable mod and are mutually exclusive:
**Blinds:** 1.12x -> 1.10x (matches flashlight)
**Spin in:** 1.05x (matches hidden)
**Traceable:** 1.05x (matches hidden)

There also are new mods with no comparison to stable (No Scope, Barrel Roll, Bubbles, Freeze Frame to name a few) and are not mutually exclusive to any stable mods. I won't comment on these for now, but discussion needs to be done on these mods before lazer scoring is finalised.

All of these values listed above may also need to be scaled so that they can appropriately be beaten with better mods. For instance, by adding HardRock to a score, you should be able to beat a score without HardRock with about a 4-5% accuracy margin, similar to how it behaves in stable.

## Mod customisation
There are a lot potential customisations. I will touch on a few:

### HalfTime
Currently, HalfTime reduces the multiplier by 0.1x for every 0.1 rate change. I suggest changing this to reduce by 0.15x for every 0.1 rate change. Lowering the speed makes it much harder to beat scores without a rate change with this system.

### Wind up/Wind down
These mods currently have flat 0.5x multipliers. I think it would be more fair if they matched how DoubleTime/HalfTime works, using the lowest of the final/initial rate to calculate the final score multiplier.

### Difficulty Adjust
This is easily the most complicated customisation mod, which currently gives a flat 0.5x multiplier, even if the customisation settings matches HardRock, which provides a positive multiplier.

There is a similar system in place mania with the key conversion mods where lowering the key count by 1 decreases the score multiplier 0.04x (starting from 0.86x for the first key below the current), but increasing the key count results in a flat 0.9x multiplier. Perhaps this system is a good precedent for how Difficulty Adjust should look like.

**Refer to previous discussions:**
https://github.com/ppy/osu/discussions/17461
https://github.com/ppy/osu/discussions/20069

## "Reading" Mods
There is a big discussion (https://github.com/ppy/osu/discussions/20476) as to whether mods like Hidden should have the same multiplier as HardRock. In my opinion, it should match stable (Hidden equals HardRock) to keep lazer leaderboards in a similar order to stable leaderboards for the time being.

## Conclusion
There is a lot more that can be said about mod multipliers, but I don't want to make this discussion 10 pages long listing every possible mod. I just hope to get the ball rolling with what I think are important changes necessary to finalise the scoring system after today's discussion.

**Comment by peppy** [comment]
One thing to note is that now that we have switched to Score V2 (and it was the same with the previous lazer scoring), mod multipliers can be changed after-the-fact relatively easily as they are applied as a final step.

So I don't think this deserves immediate priority – as in it won't be part of our push to get lazer scoring persistent this year, but it's definitely something up for discussion in the future.

**Reply by molneya** [reply]
I see where you are coming from with this, and its definitely better how mod multipliers aren't baked into the score calculation. But in my opinion, as part of getting lazer score submission online, mod multipliers need to be in a permanent state. I think it would be silly if score leaderboards could shuffle around after people have set first places with a previous mod multiplier. Again, this needs more discussion, which is partly why I made this discussion in the first place.

**Reply by peppy** [reply]
Thanks for your opinion.

**Reply by apollo-dw** [reply]
I'll add that I think it's totally fine if score leaderboards shuffle a bunch. The only thing that really needs to be done is making sure as many players expect things to be in flux, and to also make sure to announce mod multiplier updates in a very clear manner (notifications + blog posts?)

**Comment by Givikap120** [comment]
I suggest increasing DT multiplier to 1.15x (0.03x per step), because it's must be higher than FL due to the fact that it's almost always harder than FL (excluding low SR maps).
HR is almost always harder than HD, so it should be like 1.06x for HR and 1.04x for HD. Or 1.08x for HR and 1.05x for HD
Fun category mods shouldn't give any bonus score. But popular and "good" mods should probably be moved to Difficulty increase as start giving score. I think Traceable should be moved to Difficulty increase and give 0.01x less than HD.
Blinds is harder than FL so it should probably give more than FL, but I don't like the current Blinds state at all. If it gets reworked to be more pretty and less clunky - it should probably give 1.15x, as much as DT 1.5x.
Strict Tracking is arguable, cuz the map can don't have sliders at all.
Also, HT currently giving unfairly much, especially compared to EZ and NF. It should be at least -0.08x per 0.05 change (0.6x for 0.75, 0.2x for 0.5). Maybe even -0.09x.
